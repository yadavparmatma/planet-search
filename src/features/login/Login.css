.login {
    padding-top: 100px;
}

.form-signin {
    max-width: 420px;
    padding: 30px 38px 66px;
    background-color: #eee;
    margin: 100px 500px;
}

.form-signin-heading {
    text-align:center;
    margin-bottom: 30px;
}

.form-control {
    position: relative;
    font-size: 16px;
    height: auto;
    padding: 10px;
}

input[type="text"] {
    margin-bottom: 0px;
    border-bottom-left-radius: 0;
    border-bottom-right-radius: 0;
}

input[type="password"] {
    margin-bottom: 20px;
    border-top-left-radius: 0;
    border-top-right-radius: 0;
}

.colorgraph {
    height: 7px;
    border-top: 0;
    background: #c4e17f;
    border-radius: 5px;
}
