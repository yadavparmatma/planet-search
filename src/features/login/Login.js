import React, {Component} from 'react';
import {Form, Text} from 'react-form'

import './Login.css';
import SwapiService from "../../services/SwapiService";
import {authenticateUser} from "../../utils/user-login";

class Login extends Component {
    constructor(props) {
        super(props);

        this.login = this.login.bind(this);
    }

    login(character) {
        SwapiService.getCharacter(character.name).then(response => {
            const authenticatedUser = authenticateUser(response.results, character);
            authenticatedUser.length === 1 ? this.props.history.push("search") : alert("Invalid credentials");
        }).catch(e => {
            alert(`Errror: ${e.message}`)
        });
    };

    render() {
        return (
            <Form className="Login" onSubmit={this.login}>
                {({submitForm}) =>
                    (<form onSubmit={submitForm}>
                        <Text className={"userName"} placeholder={"Username"} field={"name"}/>
                        <Text className={"password"} type={"password"} placeholder={"Password"} field={"password"}/>
                        <button className="login-button" type="submit">Login</button>
                    </form>)
                }
            </Form>
        );
    }
}

export default Login;
