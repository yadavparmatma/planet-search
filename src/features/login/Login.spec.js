import React from "react";
import renderer from 'react-test-renderer';
import {Form, Text} from 'react-form'

import Login from "./Login";


describe('Login', () => {
    it('should match snapshot', async () => {
        const wrapper = renderer.create(<Login />);

        expect(wrapper.toJSON()).toMatchSnapshot();
    });
});
