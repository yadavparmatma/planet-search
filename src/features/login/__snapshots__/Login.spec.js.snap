// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should match snapshot 1`] = `
<form
  className="form-signin"
  onSubmit={[Function]}
>
  <h3
    className="form-signin-heading"
  >
    Login: using planet character
  </h3>
  <input
    className="form-control"
    onBlur={[Function]}
    onChange={[Function]}
    placeholder="Username"
    value=""
  />
  <input
    className="form-control"
    onBlur={[Function]}
    onChange={[Function]}
    placeholder="Password"
    type="password"
    value=""
  />
  <button
    className="btn btn-lg btn-primary btn-block"
    type="submit"
  >
    Login
  </button>
</form>
`;
